Translation Word Report and Rake Fix

visoft opened the pull request:
The first commit here fixes the issue with rake v12, which was introduced with the sidekiq upgrade. The solution was to update yard.

The last commit contains the changes needed to add the new Translation Word Report.

Please review @robolson

tingley commented:
While working on the next report, we realized this is filtering by project, which is outside the original requirements and probably not what's wanted.  We're going to tweak it and re-submit.

tingley commented:
Hi @robolson, this is ready for your review again.

robolson commented on app/controllers/stats_controller.rb:
```diff
@@ -74,6 +83,46 @@ def generate_commit_csv
     end
   end
 
+  def generate_translation_words_csv
```
all of this logic should live outside of the controller. Probably in a file in `/lib/reports/`

robolson commented on app/mediators/translation_update_mediator.rb:
```diff
@@ -111,6 +111,16 @@ def multi_updateable_translations_indexed_by_rfc5646_locale
 
   def update_single_translation!(translation)
+    # determine if this is the initial translation or an update based on if the
+    # tranlation has an existing value.
```
spelling error on translation

robolson commented on app/controllers/stats_controller.rb:
```diff
@@ -55,6 +55,15 @@ def index
+    respond_to do |format|
+      format.html
+      format.csv { render text: generate_translation_words_csv }
+    end
```
please split the GET of translation report HTML out from the POST for the generated CSV into two separate actions. No need to overload into 1.

robolson commented on spec/mediators/translation_update_mediator_spec.rb:
```diff
@@ -287,4 +287,17 @@
+
+      translation = FactoryGirl.build(:translation, copy: "Test", source_copy: 'Test')
+      mediator = TranslationUpdateMediator.new(translation, reviewer, ActionController::Parameters.new)
+      expect(mediator.send(:top_fuzzy_match, translation)).to eq 72.72727272727273
```
This test needs work. I have a couple suggestions:

1. Is it possible / would it make sense to test top_fuzzy_match behavior via a test of `update_single_translation!` instead? That would be preferred since `top_fuzzy_match` is supposed to be a private method.
2. calling `sleep` inside of a test is highly undesirable. Can you mock out the response from elasticsearch instead?
3. The use of 72.72727272727273 is a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) and I would like to avoid testing like this. It either needs to be explained or dropped. If we test the method via `update_single_transaction!` then we will not have this problem.

visoft commented:
I refactored the spec and added one for the `translation_date` as well. I used `sleep` only because it was used in other specs. I was able to remove it by adding `refresh: true` to the `model.import` statement in `spec_helper.rb`. 

Instead of `update_single_translation!`, which is another private method, I ended up writing the new specs against `update!`, which is public. 

Finally, I didn't mock elasticsearch because a) it would have involved a fair amount of work and b) the other specs that use ES, use the helper that I used (plus `sleep`, which we should be able to remove now). If you'd still like me to mock the ES result, just let me know.
